### ISSUE TITLE
Cannot bump dependencies in Kubernetes projects due to census-instrumentation/opencensus-proto

### ISSUE THREAD

**Issue by justaugustus** [Azure/go-autorest#413]
I've tried several combinations of versions of `azure-sdk-for-go` and `go-autorest` but have been unsuccessful in updating dependencies for the `cluster-api-provider-azure` Kubernetes project.

Here are the output logs for the current PR I have open to test (https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/216):

```
$ time make test
GOPATH=/home/augustus/go bazel run //:generate 
Starting local Bazel server and connecting to it...
INFO: Analysed target //:generate (17 packages loaded, 168 targets configured).
INFO: Found 1 target...
Target //:generate up-to-date:
  bazel-bin/generate_script
  bazel-bin/generate
INFO: Elapsed time: 21.646s, Critical Path: 0.14s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
make dep-ensure
make[1]: Entering directory '/home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure'
Repository is set up correctly
INFO: Analysed target //:dep (57 packages loaded, 7728 targets configured).
INFO: Found 1 target...
Target //:dep up-to-date:
  bazel-bin/dep_script
  bazel-bin/dep
INFO: Elapsed time: 5.101s, Critical Path: 0.28s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
make gazelle
make[2]: Entering directory '/home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure'
bazel run //:gazelle 
INFO: Analysed target //:gazelle (25 packages loaded, 124 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 2.352s, Critical Path: 0.33s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
gazelle: open /home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure/vendor/k8s.io/kubernetes/BUILD.bazel: no such file or directory
make[2]: Leaving directory '/home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure'
make[1]: Leaving directory '/home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure'
bazel build  //pkg/cloud/azure/mocks:mocks \
	//pkg/cloud/azure/services/disks/mock_disks:mocks \
	//pkg/cloud/azure/services/availabilityzones/mock_availabilityzones:mocks \
	//pkg/cloud/azure/services/groups/mock_groups:mocks \
	//pkg/cloud/azure/services/internalloadbalancers/mock_internalloadbalancers:mocks \
	//pkg/cloud/azure/services/networkinterfaces/mock_networkinterfaces:mocks \
	//pkg/cloud/azure/services/publicips/mock_publicips:mocks \
	//pkg/cloud/azure/services/publicloadbalancers/mock_publicloadbalancers:mocks \
	//pkg/cloud/azure/services/routetables/mock_routetables:mocks \
	//pkg/cloud/azure/services/securitygroups/mock_securitygroups:mocks \
	//pkg/cloud/azure/services/subnets/mock_subnets:mocks \
	//pkg/cloud/azure/services/virtualmachineextensions/mock_virtualmachineextensions:mocks \
	//pkg/cloud/azure/services/virtualmachines/mock_virtualmachines:mocks \
	//pkg/cloud/azure/services/virtualnetworks/mock_virtualnetworks:mocks
INFO: Analysed 14 targets (173 packages loaded, 2271 targets configured).
INFO: Found 14 targets...
ERROR: /home/augustus/go/src/sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent/BUILD.bazel:3:1: GoCompile vendor/contrib.go.opencensus.io/exporter/ocagent/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent.a failed (Exit 1) builder failed: error executing command 
  (cd /home/augustus/.cache/bazel/_bazel_augustus/bd8987116252386267e76c13b740e64e/sandbox/linux-sandbox/2/execroot/__main__ && \
  exec env - \
    CGO_ENABLED=1 \
    GOARCH=amd64 \
    GOOS=linux \
    GOROOT=external/go_sdk \
    GOROOT_FINAL=GOROOT \
    PATH=/usr/bin:/bin \
  bazel-out/host/bin/external/go_sdk/builder compile -sdk external/go_sdk -installsuffix linux_amd64 -src vendor/contrib.go.opencensus.io/exporter/ocagent/common.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/connection.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/nodeinfo.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/ocagent.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/options.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/transform_spans.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/transform_stats_to_metrics.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/version.go -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1.a=' -arc 'github.com/golang/protobuf/ptypes/timestamp=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/golang/protobuf/ptypes/timestamp=bazel-out/host/bin/vendor/github.com/golang/protobuf/ptypes/timestamp/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/golang/protobuf/ptypes/timestamp.a=' -arc 'go.opencensus.io=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io=bazel-out/host/bin/vendor/go.opencensus.io/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io.a=' -arc 'go.opencensus.io/plugin/ocgrpc=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/plugin/ocgrpc=bazel-out/host/bin/vendor/go.opencensus.io/plugin/ocgrpc/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/plugin/ocgrpc.a=' -arc 'go.opencensus.io/resource=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/resource=bazel-out/host/bin/vendor/go.opencensus.io/resource/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/resource.a=' -arc 'go.opencensus.io/stats=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats=bazel-out/host/bin/vendor/go.opencensus.io/stats/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats.a=' -arc 'go.opencensus.io/stats/view=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats/view=bazel-out/host/bin/vendor/go.opencensus.io/stats/view/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats/view.a=' -arc 'go.opencensus.io/tag=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/tag=bazel-out/host/bin/vendor/go.opencensus.io/tag/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/tag.a=' -arc 'go.opencensus.io/trace=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace=bazel-out/host/bin/vendor/go.opencensus.io/trace/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace.a=' -arc 'go.opencensus.io/trace/tracestate=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace/tracestate=bazel-out/host/bin/vendor/go.opencensus.io/trace/tracestate/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace/tracestate.a=' -arc 'google.golang.org/api/support/bundler=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/api/support/bundler=bazel-out/host/bin/vendor/google.golang.org/api/support/bundler/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/api/support/bundler.a=' -arc 'google.golang.org/grpc=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc=bazel-out/host/bin/vendor/google.golang.org/grpc/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc.a=' -arc 'google.golang.org/grpc/credentials=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/credentials=bazel-out/host/bin/vendor/google.golang.org/grpc/credentials/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/credentials.a=' -arc 'google.golang.org/grpc/metadata=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/metadata=bazel-out/host/bin/vendor/google.golang.org/grpc/metadata/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/metadata.a=' -o bazel-out/host/bin/vendor/contrib.go.opencensus.io/exporter/ocagent/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent.a -package_list bazel-out/host/bin/external/go_sdk/packages.txt -p sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent -- -trimpath .)
Execution platform: @bazel_tools//platforms:host_platform

Use --sandbox_debug to see verbose messages from the sandbox: builder failed: error executing command 
  (cd /home/augustus/.cache/bazel/_bazel_augustus/bd8987116252386267e76c13b740e64e/sandbox/linux-sandbox/2/execroot/__main__ && \
  exec env - \
    CGO_ENABLED=1 \
    GOARCH=amd64 \
    GOOS=linux \
    GOROOT=external/go_sdk \
    GOROOT_FINAL=GOROOT \
    PATH=/usr/bin:/bin \
  bazel-out/host/bin/external/go_sdk/builder compile -sdk external/go_sdk -installsuffix linux_amd64 -src vendor/contrib.go.opencensus.io/exporter/ocagent/common.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/connection.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/nodeinfo.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/ocagent.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/options.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/transform_spans.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/transform_stats_to_metrics.go -src vendor/contrib.go.opencensus.io/exporter/ocagent/version.go -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.a=' -arc 'github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1=bazel-out/host/bin/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1.a=' -arc 'github.com/golang/protobuf/ptypes/timestamp=sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/golang/protobuf/ptypes/timestamp=bazel-out/host/bin/vendor/github.com/golang/protobuf/ptypes/timestamp/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/golang/protobuf/ptypes/timestamp.a=' -arc 'go.opencensus.io=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io=bazel-out/host/bin/vendor/go.opencensus.io/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io.a=' -arc 'go.opencensus.io/plugin/ocgrpc=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/plugin/ocgrpc=bazel-out/host/bin/vendor/go.opencensus.io/plugin/ocgrpc/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/plugin/ocgrpc.a=' -arc 'go.opencensus.io/resource=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/resource=bazel-out/host/bin/vendor/go.opencensus.io/resource/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/resource.a=' -arc 'go.opencensus.io/stats=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats=bazel-out/host/bin/vendor/go.opencensus.io/stats/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats.a=' -arc 'go.opencensus.io/stats/view=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats/view=bazel-out/host/bin/vendor/go.opencensus.io/stats/view/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/stats/view.a=' -arc 'go.opencensus.io/tag=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/tag=bazel-out/host/bin/vendor/go.opencensus.io/tag/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/tag.a=' -arc 'go.opencensus.io/trace=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace=bazel-out/host/bin/vendor/go.opencensus.io/trace/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace.a=' -arc 'go.opencensus.io/trace/tracestate=sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace/tracestate=bazel-out/host/bin/vendor/go.opencensus.io/trace/tracestate/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/go.opencensus.io/trace/tracestate.a=' -arc 'google.golang.org/api/support/bundler=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/api/support/bundler=bazel-out/host/bin/vendor/google.golang.org/api/support/bundler/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/api/support/bundler.a=' -arc 'google.golang.org/grpc=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc=bazel-out/host/bin/vendor/google.golang.org/grpc/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc.a=' -arc 'google.golang.org/grpc/credentials=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/credentials=bazel-out/host/bin/vendor/google.golang.org/grpc/credentials/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/credentials.a=' -arc 'google.golang.org/grpc/metadata=sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/metadata=bazel-out/host/bin/vendor/google.golang.org/grpc/metadata/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/google.golang.org/grpc/metadata.a=' -o bazel-out/host/bin/vendor/contrib.go.opencensus.io/exporter/ocagent/linux_amd64_stripped/go_default_library%/sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent.a -package_list bazel-out/host/bin/external/go_sdk/packages.txt -p sigs.k8s.io/cluster-api-provider-azure/vendor/contrib.go.opencensus.io/exporter/ocagent -- -trimpath .)
Execution platform: @bazel_tools//platforms:host_platform

Use --sandbox_debug to see verbose messages from the sandbox
compile: error running compiler: exit status 2
/home/augustus/.cache/bazel/_bazel_augustus/bd8987116252386267e76c13b740e64e/sandbox/linux-sandbox/2/execroot/__main__/vendor/contrib.go.opencensus.io/exporter/ocagent/ocagent.go:299:38: undefined: "sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1".ConstantSampler_ALWAYS_ON
/home/augustus/.cache/bazel/_bazel_augustus/bd8987116252386267e76c13b740e64e/sandbox/linux-sandbox/2/execroot/__main__/vendor/contrib.go.opencensus.io/exporter/ocagent/transform_stats_to_metrics.go:52:3: unknown field 'MetricDescriptor' in struct literal of type "sigs.k8s.io/cluster-api-provider-azure/vendor/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1".Metric
INFO: Elapsed time: 3.869s, Critical Path: 0.70s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
Makefile:165: recipe for target 'generate' failed
make: *** [generate] Error 1

real	0m56.523s
user	0m12.150s
sys	0m13.008s
```

Let me know if I can provide additional information!

cc: @lachie83 @craiglpeters @ritazh @khenidak

**Comment by jhendrixMSFT** [comment]
Hello!  Is the ask here for us to upgrade our version of `census-instrumentation/opencensus-proto`?

**Comment by justaugustus** [comment]
@jhendrixMSFT -- Not quite? The last versions that worked for us were azure-sdk-for-go `v27.1.0` and go-autorest `v11.5.2`. I believe _something_ in >= v11.6.0 introduced a transitive dependency on opencensus, which doesn't play nice with the rest of our project.

**Comment by justaugustus** [comment]
Primarily, we depend on the compute, network, and resources APIs, but I know the cloud controller manager also depends on the storage API

**Comment by jhendrixMSFT** [comment]
The dependency on opencensus has been there since v11.2.0, however there were some breaking changes in one of its dependencies that broke us and was fixed in v11.6.1 (see https://github.com/Azure/go-autorest/issues/375).  I suspect this is the root of the problem although I haven't dug further into it.

**Comment by jhendrixMSFT** [comment]
Looking at your PR it appears that `opencensus-proto` wasn't updated to v0.2.0, i.e. it's missing changes to trace_config.pb.go which is where the missing enum is defined.  Do you know why?

**Comment by justaugustus** [comment]
@jhendrixMSFT -- working on gathering more evidence for you!
ref: https://github.com/kubernetes/kubernetes/pull/79574

**Comment by tamalsaha** [comment]
I just spent last 3-4 hours stuck on this issue. Our dependencies also come via Kubernetes. The problem that I see is this:

Originally we were using the following versions:
```
require (
	github.com/Azure/azure-sdk-for-go v21.3.0
	github.com/Azure/go-autorest v11.1.0
)
``` 

Then we updated to the following versions in May 2019 without any problem:
```
	github.com/Azure/azure-sdk-for-go v29.0.0+incompatible
	github.com/Azure/go-autorest v12.0.0+incompatible
``` 

This library [started using leaf modules](https://github.com/Azure/go-autorest/pull/386) between v12.0.0 and v12.0.1 release. So, today when I was running `go mod tidy`, `go mod vendor` will complete but the `tracing` sub package will be missing. At one point I was getting the error that `github.com/Azure/go-autorest/tracing` package was defined by multiple modules.  

Currently I have fallback to the [following combination](https://github.com/pharmer/cloud/pull/18/files) which makes everything compile in my case:
```
require (
	github.com/Azure/azure-sdk-for-go v29.0.0+incompatible
	github.com/Azure/go-autorest v11.1.2+incompatible
)
```
But my understanding is that this is the wrong version of autorest for the matching azure-sdk-go.

I tried to write a standalone program https://github.com/tamalsaha/azurestuff to see what go mod does in isolation. This is what it does:

```
require (
	github.com/Azure/go-autorest/autorest v0.5.0
	github.com/Azure/go-autorest/tracing v0.1.0
)
```

Frankly, I don't know what are the `versions` for the leaf modules? Does that come from the git tag? I tried that and it does not work.

```
$ go mod tidy
go: finding github.com/Azure/go-autorest/autorest v12.3.0
go: finding github.com/Azure/go-autorest/tracing v12.3.0
go: errors parsing go.mod:
/home/tamal/go/src/github.com/tamalsaha/azurestuff/go.mod:6: invalid module version "v12.3.0": unknown revision autorest/v12.3.0
/home/tamal/go/src/github.com/tamalsaha/azurestuff/go.mod:7: invalid module version "v12.3.0": unknown revision tracing/v12.3.0

```

I appreciate any help fixing this mess.

**Comment by tamalsaha** [comment]
I found how to fix the problems I was facing. I have opened a pr to add instructions for that. https://github.com/Azure/go-autorest/pull/431

**Comment by jhendrixMSFT** [comment]
All of the large version tags (e.g. v12.3.0) are for non-module, monolithic releases of this repo; it's to support consumers that use tools like `dep` or `glide` for dependency management.
This repo defines the following modules.
- autorest/adal
- autorest/azure/cli
- autorest/date
- autorest/mocks
- autorest/to
- autorest/validation
- autorest
- logger
- tracing

I need more info to help diagnose the root of your problem as I can successfully build a simple program that consumes `v31.1.0+incompatible` of azure-sdk-for-go along with `autorest/v0.5.0`.

**Comment by tamalsaha** [comment]
@jhendrixMSFT , here is an example of the error we are seeing:

```
cannot load github.com/Azure/go-autorest/tracing: ambiguous import: found github.com/Azure/go-autorest/tracing in multiple modules:
        github.com/Azure/go-autorest v11.7.0+incompatible (/home/fahim/go/pkg/mod/github.com/!azure/go-autorest@v11.7.0+incompatible/tracing)
        github.com/Azure/go-autorest/tracing v0.1.0 (/home/fahim/go/pkg/mod/github.com/!azure/go-autorest/tracing@v0.1.0)
```

**Comment by jhendrixMSFT** [comment]
What's causing both of these packages to be added to your go.mod file?

**Comment by tamalsaha** [comment]
`github.com/Azure/go-autorest@v11.7.0` is coming from Kubernetes which my project depends on. And I am trying to use `github.com/Azure/azure-sdk-for-go v31.0.0+incompatible` directly. 

https://github.com/Azure/go-autorest/pull/431 (specifically the replacement) solved the problem for me.

**Comment by jhendrixMSFT** [comment]
But what is causing `tracing/v0.1.0` to be added to your go.mod?  `go-autorest@v11.7.0` already includes the `tracing` package.  Is it the consumption of `azure-sdk-for-go@v31.0.0+incompatible` that's causing it?

**Comment by tamalsaha** [comment]
> Is it the consumption of azure-sdk-for-go@v31.0.0+incompatible that's causing it?

yes.

**Comment by tamalsaha** [comment]
I run `go mod tidy`, then go does its thing. That adds the tracing package.

**Comment by tamalsaha** [comment]
@jhendrixMSFT , to see this problem, you can clone pharmer/cloud repo and follow the instructions here:

https://github.com/pharmer/cloud/commit/01456392c26d0b7d548187040e5c4a71a51bb306

You will see that either go complains about ambiguity or does not vendor correctly.

I am using go 1.12.7

**Comment by alexeldeib** [comment]
I'm building via Bazel and started seeing different kinds of issues here as well. Vendoring solves the problem. Might be an issue with bazel/gazelle, but figured I'd add that anecdata.

**Comment by jhendrixMSFT** [comment]
This should now be fixed in v12.4.1, module tracing/v0.2.0.
